.github/workflows/branch_off.yml

```diff
@@ -0,0 +1,88 @@
+name: Branch off
+
+on:
+  workflow_dispatch:
+  workflow_call:
+    outputs:
+      bump_version_pr_url:
+        description: 'URL of the bump version PR'
+        value: ${{ jobs.branch-off.outputs.bump_version_pr_url }}
+env:
+  PYTHON_VERSION: "3.9"
+
+jobs:
+  branch-off:
+    runs-on: ubuntu-latest
+    outputs:
+      bump_version_pr_url: ${{ steps.create-pr.outputs.pull-request-url }}
+
+    steps:
+      - name: Checkout this repo
+        uses: actions/checkout@v6
+        with:
+          ref: main
+
+      - name: Define all versions
+        id: versions
+        shell: bash
+        run: |
+          # example: 2.20.0-rc0 in VERSION.txt -> 2.20
+          echo "current_version_major_minor=$(cut -d "." -f 1,2 < VERSION.txt)" >> "$GITHUB_OUTPUT"
+          # example: 2.20.0-rc0 in VERSION.txt -> 2.21.0-rc0
+          echo "next_version_rc0=$(awk -F. '/[0-9]+\./{$2++;print}' OFS=. < VERSION.txt)" >> "$GITHUB_OUTPUT"
+
+      - name: Create release branch and tag
+        shell: bash
+        env:
+          # We use the HAYSTACK_BOT_TOKEN here so the PR created by the step will
+          # trigger required workflows and can be merged by anyone
+          GITHUB_TOKEN: ${{ secrets.HAYSTACK_BOT_TOKEN }}
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+
+          # Create the release branch from main
+          git checkout -b v${{ steps.versions.outputs.current_version_major_minor }}.x
+          git push -u origin v${{ steps.versions.outputs.current_version_major_minor }}.x
```

Comment:
Maybe we've already discussed about this (sorry if yes), but do you think it would worth checking if the branch already exists here?